client/js/s3/request-signer.js

```diff
@@ -265,7 +268,9 @@ qq.s3.RequestSigner = function(o) {
             pendingSignatureData = pendingSignatures[id],
             promise = pendingSignatureData.promise,
             signatureConstructor = pendingSignatureData.signatureConstructor,
-            errorMessage, response;
+            oldCredentials = Object.assign({}, credentialsProvider.get()),
```

Comment:
You should probably just refer to [the `qq.extend` function in utils](http://docs.fineuploader.com/branch/master/api/qq.html#qq.extend).